### PR TITLE
dts: nordic: simplify definition of flash and sram sizes in dts

### PR DESCRIPTION
**Pull request by ioannisg**
We only need to add the reg property in flash0 and sram0,
in the different DTS headers for the nRF SOCS. We do not
seem to need to define the nodes again. This commit applies
this simplification for flash and sram sizes definition.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

### CODE REVIEW

**Review by carlescufi — APPROVED** [review]

**Review by anangl — APPROVED** [review]

**Comment by carlescufi** [comment]
recheck

**Comment by nvlsianpu** [comment]
pleas wait for my approval.

**Review by nvlsianpu — APPROVED** [review]

**Pull request merged**
